### PR TITLE
Feature: Payment Gateways Settings migration

### PR DESCRIPTION
**Pull request by pauloiankoski**
## Description

This PR adds a new database migration copying the current payment gateways settings from v2 to v3:
```
gateways > gateways_v3
gateways_label > gateways_label_v3
default_gateway > default_gateway_v3
```

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205478875158267

### CODE REVIEW

File: src/PaymentGateways/ServiceProvider.php [hunk 1]
```diff
@@ -65,12 +66,14 @@ public function boot()
...
 
     /**
+     * @unreleased add CopyV2GatewaysSettingsToV3 migration
      * @since 2.33.0 add RemovePaymentIntentSecretMeta migration
      * @since 2.19.6
      */
     private function registerMigrations()
     {
         give(MigrationsRegister::class)->addMigrations([
+            CopyV2GatewaysSettingsToV3::class,
```
Comment (jonwaldstein) [inline src/PaymentGateways/ServiceProvider.php:76]: I think the migrations runs in order - so I would put this last in the list.

File: src/PaymentGateways/Migrations/CopyV2GatewaysSettingsToV3.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        return strtotime('2023-09-11');
+    }
+
+    /**
+     * @inerhitDoc
+     */
+    public function run()
+    {
+        $v2Gateways = give_get_option('gateways', []);
+        give_update_option('gateways_v3', $v2Gateways);
```
Comment (jonwaldstein) [inline src/PaymentGateways/Migrations/CopyV2GatewaysSettingsToV3.php:35]: Shouldn't this be filtering out any non v3 compatible gateways?

Comment (pauloiankoski) [reply src/PaymentGateways/Migrations/CopyV2GatewaysSettingsToV3.php]: Updated on 68eab7d6bfac871bc38e81d58f430c81d9c74d75

File: src/PaymentGateways/ServiceProvider.php (hunk 1 above)
Comment (pauloiankoski) [reply src/PaymentGateways/ServiceProvider.php]: Updated on 707e8bbff7a529210ccbc63e2f80ab42756625fe

**Review by jonwaldstein — APPROVED** [review]
Looks good! Let's get this one to QA 🧙

**Review by rickalday — APPROVED** [review]
Passed manual QA tests

**Pull request merged**